main.js

```diff
@@ -0,0 +1,165 @@
+
+var setDefaultLocation = function() {
+  //if zipcode, store the zipcode
+  //if no zipcode, store the lat/long
+  var defaultValues = {};
+  if (searchLocation.zip){
+    defaultValues.zip = searchLocation.zip;
+  } else {
+    defaultValues.latitude = searchLocation.latitude;
+    defaultValues.longitude = searchLocation.longitude;
+  }
+  var defaultLocation = JSON.stringify(defaultValues);
+  localStorage.defaultLocation = defaultLocation;
+}
+
+var getDefaultLocation = function() {
+  return localStorage.defaultLocation ? JSON.parse(localStorage.defaultLocation) : {};
+}
+
+var searchLocation = getDefaultLocation();
```

Comment:
This line should be in some kind of initialization function or start function so that it is clear what the initial behavior is.